### PR TITLE
GH-1682 [sql_alchemy] adding get_logs() function

### PR DESCRIPTION
**Pull request by agl29**
## What changes were proposed in this pull request?

- get_logs() function in sql_alchemy

### CODE REVIEW

**Comment by agl29** [comment]
But this is giving error (ss).
Screenshot 2021-02-10 at 9 10 54 PM

**Comment by agl29** [comment]
But when i am printing `result.cursor.fetch_logs()` at https://github.com/cloudera/hue/blob/master/desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py#L255. it is giving the right logs as below.
` ['INFO  : Compiling command(queryId=hive_20210210142934_3779800e-55bc-4414-81cc-90455bf0670b): SELECT *', 'FROM default.sample_07', 'LIMIT 100', '', 'INFO  : Semantic Analysis Completed (retrial = false)', 'INFO  : Created Hive schema: Schema(fieldSchemas:[FieldSchema(name:sample_07.code, type:string, comment:null), FieldSchema(name:sample_07.description, type:string, comment:null), FieldSchema(name:sample_07.total_emp, type:int, comment:null), FieldSchema(name:sample_07.salary, type:int, comment:null)], properties:null)', 'INFO  : Completed compiling command(queryId=hive_20210210142934_3779800e-55bc-4414-81cc-90455bf0670b); Time taken: 0.053 seconds', 'INFO  : Executing command(queryId=hive_20210210142934_3779800e-55bc-4414-81cc-90455bf0670b): SELECT *', 'FROM default.sample_07', 'LIMIT 100', '', 'INFO  : Completed executing command(queryId=hive_20210210142934_3779800e-55bc-4414-81cc-90455bf0670b); Time taken: 0.008 seconds', 'INFO  : OK']`

but when I want to access them in get_logs() function it is giving error as described above.

**Comment by agl29** [comment]
Although we can use global var for getting corsor.fetch_logs() in get_logs() function but I don't think that it is good idea.

File: desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py [hunk 1]
```diff
@@ -394,8 +394,12 @@ def cancel(self, notebook, snippet):
 
   @query_error_handler
   def get_log(self, notebook, snippet, startFrom=None, size=None):
-    return ''
+    guid = snippet['result']['handle']['guid']
+    connection = CONNECTIONS.get(guid)
+    result = connection['result']
+    cursor = result.cursor
```
Comment (romainr) [inline desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py:400]: combine?

Comment (romainr) [reply desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py:400]: Also check if connection, cursor are there...
like in progress()

File: desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py [hunk 2]
```diff
@@ -394,8 +394,12 @@ def cancel(self, notebook, snippet):
...
 
+    return cursor.fetch_logs()
```
Comment (romainr) [inline desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py:402]: Can we provide an offset in the sqlalchemy API?

**Review by romainr — COMMENTED** [review]
1. In your screenshot, have you check that the handle/guid etc are valid?

2. "GH-1682 [sql_alchemy get_logs()] adding get_logs() function" --> "GH-1682 [sqlalchemy] adding get_logs() function" Please simpler name?

3. Ant unit test?

**Comment by agl29** [comment]
> 1. In your screenshot, have you check that the handle/guid etc are valid?


Screenshot 2021-02-11 at 12 15 40 AM

I checked with hive query execute API output that guid is same.

**Comment by agl29** [comment]
> > 1. In your screenshot, have you check that the handle/guid etc are valid?
> 
> Screenshot 2021-02-11 at 12 15 40 AM
> 
> I checked with hive query execute API output that guid is same.

is something else I have to look particularly @romainr ?

**Comment by romainr** [comment]
Let's fix the other comments and push it (will be easier to debug/understand)!

**Comment by agl29** [comment]
Screenshot 2021-02-15 at 4 53 01 PM
local terminal output with sqlalchemy for the statement 'SELECT 1'

File: desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py [hunk 3]
```diff
@@ -254,6 +254,7 @@ def execute(self, notebook, snippet):
     result = connection.execute(statement)
 
     cache = {
+      'logs': [message for message in result.cursor.fetch_logs()] if result.cursor and hasattr(result.cursor, 'fetch_logs') else [],
```
Comment (romainr) [inline desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py:257]: Move it one line up into a variable? (easier to understand)

**Review by romainr — APPROVED** [review]

File: desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py (hunk 3 above)
Comment (agl29) [reply desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py]: done

**Pull request merged**